pkg/scheduler/plugins/nodereservation/plugin.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodereservation
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	resschedplug "k8s.io/kubernetes/pkg/scheduler/framework/plugins/noderesources"
+
+	apiext "github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/util"
+)
+
+const (
+	Name              = "NodeReservation"
+	preFilterStateKey = "PreFilter" + Name
+)
+
+var (
+	_ framework.FilterPlugin    = &Plugin{}
+	_ framework.PreFilterPlugin = &Plugin{}
+	//_ framework.ScorePlugin = &Plugin{}
```

Comment:
We should implement the score plugin, but can implement later.  Please add TODO here.